service/src/test/java/uk/gov/hmcts/reform/fpl/service/validators/CaseSubmissionGuardTest.java

```diff
@@ -0,0 +1,151 @@
+package uk.gov.hmcts.reform.fpl.service.validators;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.enums.Event.ALLOCATION_PROPOSAL;
+import static uk.gov.hmcts.reform.fpl.enums.Event.APPLICANT;
+import static uk.gov.hmcts.reform.fpl.enums.Event.CASE_NAME;
+import static uk.gov.hmcts.reform.fpl.enums.Event.CHILDREN;
+import static uk.gov.hmcts.reform.fpl.enums.Event.DOCUMENTS;
+import static uk.gov.hmcts.reform.fpl.enums.Event.GROUNDS;
+import static uk.gov.hmcts.reform.fpl.enums.Event.HEARING_NEEDED;
+import static uk.gov.hmcts.reform.fpl.enums.Event.ORDERS_NEEDED;
+import static uk.gov.hmcts.reform.fpl.enums.Event.RESPONDENTS;
+
+@ExtendWith(SpringExtension.class)
+class CaseSubmissionGuardTest {
+
+    @Mock
+    private EventChecker eventChecker;
+
+    @InjectMocks
+    private CaseSubmissionGuard caseSubmissionValidator;
+
+    private final CaseData caseData = CaseData.builder().build();
+
+    @Test
+    void shouldReportGroupedErrorsForAllRelevantEvents() {
+        final List<String> caseNameErrors = List.of("Case name error1");
```

Comment:
spacing of `error1` just to be consistent with the rest

Reply:
done